src/Control/Monad/Trans/Instances.hs

```diff
@@ -1248,3 +1252,129 @@ data C1_0'SelectT
 #  endif
 # endif
 #endif
+
+#if !(MIN_VERSION_base(4,9,0))
+#if MIN_VERSION_transformers(0,4,0) && !(MIN_VERSION_transformers(0,5,0))
```

Comment:
This particular combination of CPP is common enough that there is a `TRANSFORMERS_FOUR` macro defined for it:

```suggestion
# if defined(TRANSFORMERS_FOUR)
```